### PR TITLE
Migrate java args scope

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [/] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [/] I have not broken the cheatsheet

### CODE REVIEW

File: data/fixtures/scopes/java/argument.actual.iteration.scope [hunk 1]
```diff
@@ -0,0 +1,10 @@
+myFunk(5, "hello");
+---
+
+[Range] = 0:7-0:17
+         >----------<
+0| myFunk(5, "hello");
+
+[Domain] = 0:6-0:18
+        >------------<
+0| myFunk(5, "hello");
```
Comment (pokey) [inline data/fixtures/scopes/java/argument.actual.iteration.scope:10]: I don't believe this is what we usually do, is it? I think we pretty much universally exclude the delimiter pair from iteration scope, but maybe I'm wrong?

File: data/fixtures/scopes/java/argument.actual.scope
Comment (pokey) [inline data/fixtures/scopes/java/argument.actual.scope:1]: shall we add a multi-line test too to make sure it's getting the right insertion delimiter?

File: data/fixtures/scopes/java/argument.formal.iteration.scope [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+}
+---
+
+[Range] = 1:23-1:45
+                         >----------------------<
+1|     public void myFunk(int value, String name) { }
+
+[Domain] = 1:4-1:50
+      >----------------------------------------------<
+1|     public void myFunk(int value, String name) { }
```
Comment (pokey) [inline data/fixtures/scopes/java/argument.formal.iteration.scope:12]: this is nice. do we do this in typescript? we should prob document that this is the desired behaviour in the scope info if we don't

File: data/fixtures/scopes/java/argument.formal.scope
Comment (pokey) [inline data/fixtures/scopes/java/argument.formal.scope:1]: as above, multi-line test might be nice

File: packages/cursorless-engine/src/languages/getNodeMatcher.ts [hunk 3]
```diff
@@ -54,7 +53,6 @@ export const languageMatchers: Record<
   csharp,
   css: scss,
   go,
-  java,
```
Comment (pokey) [inline packages/cursorless-engine/src/languages/getNodeMatcher.ts:57]: 🥳

File: queries/java.scm [hunk 4]
```diff
@@ -326,3 +326,43 @@
...
+(
+  (formal_parameters
+    (_)? @_.leading.endOf
+    .
+    (_) @argumentOrParameter
+    .
+    (_)? @_.trailing.startOf
+  ) @_dummy
+  (#not-type? @argumentOrParameter "block_comment")
+  (#single-or-multi-line-delimiter! @argumentOrParameter @_dummy ", " ",\n")
```
Comment (pokey) [inline queries/java.scm:341]: won't this lose indentation?

File: queries/java.scm [hunk 5]
```diff
@@ -326,3 +326,43 @@
...
+;;!      ^^^^^^^^^^^^
+(
+  (formal_parameters
+    (_)? @_.leading.endOf
+    .
+    (_) @argumentOrParameter
+    .
+    (_)? @_.trailing.startOf
+  ) @_dummy
+  (#not-type? @argumentOrParameter "block_comment")
```
Comment (pokey) [inline queries/java.scm:340]: might be worth folding a comment into a test?

**Review by pokey — COMMENTED** [review]
another one bites the dust! left a bunch of minor comments

File: data/fixtures/scopes/java/argument.actual.iteration.scope (hunk 1 above)
Comment (AndreasArvidsson) [reply data/fixtures/scopes/java/argument.actual.iteration.scope:10]: https://github.com/cursorless-dev/cursorless/blob/189813994fbe5a4a482f49b481427a8937ec781c/data/fixtures/scopes/javascript.core/argument.actual.iteration.scope

Comment (pokey) [reply data/fixtures/scopes/java/argument.actual.iteration.scope:10]: oh wow super weird. shall we just leave it in both places then? I could go either way, but agree consistency is most important

File: data/fixtures/scopes/java/argument.actual.scope
Comment (AndreasArvidsson) [reply data/fixtures/scopes/java/argument.actual.scope:1]: done

File: data/fixtures/scopes/java/argument.formal.iteration.scope (hunk 2 above)
Comment (AndreasArvidsson) [reply data/fixtures/scopes/java/argument.formal.iteration.scope:12]: Yes we do
https://github.com/cursorless-dev/cursorless/blob/fd229a29fd36fbfb33c4e2d87afd88fad24ae64b/data/fixtures/scopes/javascript.core/argument.formal.iteration.scope

Description updated

File: data/fixtures/scopes/java/argument.formal.scope
Comment (AndreasArvidsson) [reply data/fixtures/scopes/java/argument.formal.scope:1]: done

File: queries/java.scm (hunk 4 above)
Comment (AndreasArvidsson) [reply queries/java.scm:341]: This is the same as we do in javascript. Our code takes care of indentation matching whenever you insert a new line.

File: data/fixtures/scopes/java/argument.actual.iteration.scope (hunk 1 above)
Comment (AndreasArvidsson) [reply data/fixtures/scopes/java/argument.actual.iteration.scope:10]: I think we should leave it for now and then later we might actually evaluate the benefits of doing this and make all languages consistent? It's not like either of us have had a problem with this on javascript?

File: queries/java.scm (hunk 4 above)
Comment (pokey) [reply queries/java.scm:341]: Well aren't we just fancy then 💅

File: queries/java.scm (hunk 5 above)
Comment (pokey) [reply queries/java.scm:340]: Yeah probably worth discussing, but out of scope for this PR

File: data/fixtures/scopes/java/argument.actual2.scope
Comment (pokey) [inline data/fixtures/scopes/java/argument.actual2.scope:1]: This format has really turned out to be a big win. Super readable and compact at the same time

**Review by pokey — APPROVED** [review]
🚀

File: queries/java.scm (hunk 5 above)
Comment (AndreasArvidsson) [reply queries/java.scm:340]: Agreed

File: queries/java.scm (hunk 4 above)
Comment (AndreasArvidsson) [reply queries/java.scm:341]: Oh yes :D

File: data/fixtures/scopes/java/argument.actual2.scope
Comment (AndreasArvidsson) [reply data/fixtures/scopes/java/argument.actual2.scope:1]: Total agree. Score one for custom formats  :D

**Pull request merged**